### PR TITLE
Moved HUI client over from `kat-usage-reports`

### PR DESCRIPTION
**Pull request by dapenguin**

### CODE REVIEW

File: lib/huiClient.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+function getUsage(licenceId, aggregation, startDate, endDate, type = '', filter = '') {
+  let start = '';
+  let end = '';
+  let url = `${huiBasePath}${licencePath}${licenceId}?aggregation=${aggregation}`;
+
+  // Get the prev week/month data too. We make a clone of the start date because
+  // moments are mutable.
+  const adjustedStartDate = startDate.clone().startOf(aggregation === 'week' ? 'isoweek' : aggregation).subtract(1, aggregation);
+
+  const dateFormat = aggregation === 'week' ? '[wk]W/GGGG' : '[m]M/GGGG';
```
Comment (ciprianlujeru) [inline lib/huiClient.js:19]: This looks good but, previously, the week start/end dates were created based on isoWeek (and year). Not sure if this results in the same dates.
https://github.com/Financial-Times/kat-usage-report/blob/master/server/lib/hui.js#L66

File: lib/huiClient.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+const log = require('@financial-times/n-logger').default;
+
+const huiBasePath = process.env.HUI_BASE_PATH;
```
Comment (ciprianlujeru) [inline lib/huiClient.js]: Maybe get these vars from the config.

File: lib/huiClient.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  // adding type (device || platform) query param to the URL
+  if (type !== '') {
+    url += `&${type}=${encodeURIComponent(filter)}`;
+  }
+
+  log.debug({operation: 'getUsage', licenceId, aggregation, start, end, url});
+
+  return fetch(url, options)
```
Comment (ciprianlujeru) [inline lib/huiClient.js:35]: All the other client proxies use `fetch-retry-or-die` as fetch. This doesn't seem like it does.

File: lib/huiClient.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+const log = require('@financial-times/n-logger').default;
+
+const huiBasePath = process.env.HUI_BASE_PATH;
+const licencePath = '/licences/';
+const options = {
```
Comment (ciprianlujeru) [inline lib/huiClient.js]: If you're going to use `fetch-retry-or-die` (as mentioned below). You might want to extend `config.fetchOptions` (as all the other client proxies do - https://github.com/Financial-Times/kat-client-proxies/blob/master/lib/accessLicenceClient.js#L11)

File: lib/huiClient.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+  // adding type (device || platform) query param to the URL
+  if (type !== '') {
+    url += `&${type}=${encodeURIComponent(filter)}`;
+  }
+
+  log.debug({operation: 'getUsage', licenceId, aggregation, start, end, url});
+
+  return fetch(url, options)
+    .then((data) => data.json())
```
Comment (ciprianlujeru) [inline lib/huiClient.js]: The other proxies use `helpers.parseJsonRes` in order to parse the response.

File: lib/huiClient.js (hunk 1 above)
Comment (dapenguin) [reply lib/huiClient.js:19]: Yeah, saw that as well. This code uses the ISO Week (the capital `W` in the `dateFormat`) and Year (`GGGG`), as per the MomentJS docs: https://momentjs.com/docs/#/displaying/, so it should be ok.

Comment (ciprianlujeru) [reply lib/huiClient.js:19]: :+1:

**Pull request merged**